frontend/src/contexts/authStateContext.tsx

```diff
@@ -32,9 +29,10 @@ export const AuthStateContextProvider = (props: { children?: any }) => {
   return (
     <AuthStateContext.Provider
       value={{
-        authenticated: keycloak.obj?.authenticated,
-        ready: keyCloakReady,
-        userInfo,
+        // if user info is not available when authenticated, then the auth state is not ready
```

Comment:
Cool, moving this makes more sense to me.

Reply:
I also like that we are using React `Context` for this state as it need to be "globally" available to most (if not all) of our components.

What I'd like to suggest is to try to refactor the Keycloak hook and this Context into one solution/provider to simplify how we handle things. Right now this context needs to know about keycloak internals anyway, and then the app components need to be aware of this context "ready" value and also the "keycloak" object returned from the hook. I rather have everything under one provider - either the hook or the context.

This shouldn't block this PR from merging. We can tackle that later as tech-debt.

Good work!